templates/CRM/Civicase/Page/ContactActivityTab.tpl

```diff
@@ -1,4 +1,9 @@
-<div id="civicaseActivitiesTab" >
+<div id="civicaseActivitiesTab">
+  {if $action EQ 16 and $permission EQ 'edit' and !$addAssigneeContact and !$addTargetContact}
```

Comment:
I'm wondering if we really do need all this checks at all? 
Can't we just include the template. 
I would suggest we remove all these checks and include the template directly. 
We might have issues later when this activity links will behave differently that the one in Case details because those ones are displayed without these extra conditions.

Reply:
I think we should have these. Because the New Activity button here is supposed to work exactly same as the Core. If we dont have permission, some one without the permission to create activities will be able to create the Activities.

Reply:
Yeah, We need to check the other links so that we have some sort of consistency. When we are ready to do that ,will be easier to extract the permission logic somewhere so that these two places can use it.

I am just concerned about copying code from core because indirectly we are introducing a bottleneck in our own code. When something changes in core in some of the function we copied, it could potentially break our code too.